src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -16,222 +16,75 @@
+	 * 1177-1184).
 	 *
 	 */
-	bool init_randomcoefficients();
+class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>> {
+public:
+		/** default constructor */
+		RandomFourierGaussPreproc();
 
-	/** a getter
-	 * @return the set value of protected member dim_input_space
-	 */
-	int32_t get_dim_input_space() const;
+		~RandomFourierGaussPreproc();
 
-	/** a getter
-	 * @return the set value of protected member dim_feature_space
-	 */
-	int32_t get_dim_feature_space() const;
+		virtual SGVector<float64_t> apply_to_feature_vector(SGVector<float64_t> vector) override;
 
-	/** inherited from base class
-	 * does nothing
-	 */
-	void cleanup();
+		/** @return C_DENSE*/
+		virtual EFeatureType get_feature_type() override;
 
-	/// return the name of the preprocessor
-	virtual const char* get_name() const { return "RandomFourierGaussPreproc"; }
+		/** @return F_DREAL */
+		virtual EFeatureClass get_feature_class() override;
 
-	/// return a type of preprocessor
-	virtual EPreprocessorType get_type() const { return P_RANDOMFOURIERGAUSS; }
+		virtual void fit(std::shared_ptr<Features> f) override;
 
-protected:
-	/** default processing routine, inherited from base class
-	 * @param matrix the features matrix to be processed
-	 * @return the processed feature matrix from the DenseFeatures<float64_t>
-	 * class in case (2) (see description above) this routine requires only
-	 * steps 2a) and 2b), the rest is determined automatically
-	 */
-	virtual SGMatrix<float64_t> apply_to_matrix(SGMatrix<float64_t> matrix);
+		/** does nothing */
+		void cleanup();
 
-	/**
-	 * helper for copy constructor and assignment operator=
-	 */
-	void copy(const RandomFourierGaussPreproc & feats); // helper for two constructors
+		virtual const char* get_name() const
+		{
+			return "RandomFourierGaussPreproc";
+		}
 
+		virtual EPreprocessorType get_type() const
+		{
+			return P_RANDOMFOURIERGAUSS;
+		}
 
-	/** dimension of input features
-	 * width of gaussian kernel in the form of exp(-x^2 / (2.0 kernelwidth^2) ) NOTE the 2.0 and the power ^2 !
-	 */
-	float64_t kernelwidth;
+	protected:
 
-	/** dimension of input features
-	 * width of gaussian kernel in the form of exp(-x^2 / (2.0 kernelwidth^2) ) NOTE the 2.0 and the power ^2 !
-	 */
-	float64_t cur_kernelwidth;
+		virtual SGMatrix<float64_t> apply_to_matrix(SGMatrix<float64_t> matrix) override;
 
-	/** desired dimension of input features as set by void set_dim_input_space(const int32_t dim)
-	 *
-	 */
-	int32_t dim_input_space;
+		virtual void sample_spectral_density(int32_t dim_input_space);
 
-	/** actual dimension of input features as set by bool init_randomcoefficients() or void set_randomcoefficients
-	 *
-	 */
-	int32_t cur_dim_input_space;
+		/** Helper method which generates random coefficients and stores in the
+		 *  internal members.
+		 *
+		 * @param dim_input_space input space dimension
+		 */
+		virtual void init_basis(int32_t dim_input_space);
 
+		/** width */
```

Comment:
again: let's not have the useless docstrings, the name containts everything that the comment does so the comment is not useful